### ISSUE TITLE
History Refresh Error v9.1.3.0

### ISSUE THREAD

**Issue by ch3l4z** [EDDiscovery/EDDiscovery#1760]
EDDiscovery: v9.1.3.0
ED: 2018.02.12 165575

Updated to latest release after experiencing the same issue as #1756 
Ran ok for a bit and then a new Exception (sorry,lost clipboard, cant paste exception)
Restarted EDD and bumped into History Refresh Error:

History Refresh Error: System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument, ExceptionResource resource)
   at EliteDangerousCore.StarScan.ProcessElements(JournalScan sc, ISystem sys, SystemNode sn, String customname, List`1 elements, ScanNodeType starscannodetype, Boolean isbeltcluster) in C:\Code\EDDiscovery\EliteDangerous\EliteDangerous\StarScan.cs:line 736
   at EliteDangerousCore.StarScan.Process(JournalScan sc, ISystem sys, Boolean reprocessPrimary) in C:\Code\EDDiscovery\EliteDangerous\EliteDangerous\StarScan.cs:line 323
   at EliteDangerousCore.HistoryList.ProcessUserHistoryListEntries(Func`2 hlfilter) in C:\Code\EDDiscovery\EliteDangerous\HistoryList\HistoryList.cs:line 946
   at EliteDangerousCore.HistoryList.LoadHistory(EDJournalClass journalmonitor, Func`1 cancelRequested, Action`2 reportProgress, String NetLogPath, Boolean ForceNetLogReload, Boolean ForceJournalReload, Int32 CurrentCommander, Boolean Keepuievents) in C:\Code\EDDiscovery\EliteDangerous\HistoryList\HistoryList.cs:line 915
   at EDDiscovery.EDDiscoveryController.DoRefreshHistory(RefreshWorkerArgs args) in C:\Code\EDDiscovery\EDDiscovery\EDDiscoveryController.cs:line 811

./Ch3l4z

**Comment by Removerboy** [comment]
Same here! I got a UI error (forgive me for not copying that specific program exception) This is the error I got afterwards:

EDDiscovery: 9.1.3.0
Elite Dangerous version: 3.0.1.401 EDH

History Refresh Error: System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument, ExceptionResource resource)
   at EliteDangerousCore.StarScan.ProcessElements(JournalScan sc, ISystem sys, SystemNode sn, String customname, List`1 elements, ScanNodeType starscannodetype, Boolean isbeltcluster) in C:\Code\EDDiscovery\EliteDangerous\EliteDangerous\StarScan.cs:line 736
   at EliteDangerousCore.StarScan.Process(JournalScan sc, ISystem sys, Boolean reprocessPrimary) in C:\Code\EDDiscovery\EliteDangerous\EliteDangerous\StarScan.cs:line 323
   at EliteDangerousCore.HistoryList.ProcessUserHistoryListEntries(Func`2 hlfilter) in C:\Code\EDDiscovery\EliteDangerous\HistoryList\HistoryList.cs:line 946
   at EliteDangerousCore.HistoryList.LoadHistory(EDJournalClass journalmonitor, Func`1 cancelRequested, Action`2 reportProgress, String NetLogPath, Boolean ForceNetLogReload, Boolean ForceJournalReload, Int32 CurrentCommander, Boolean Keepuievents) in C:\Code\EDDiscovery\EliteDangerous\HistoryList\HistoryList.cs:line 915
   at EDDiscovery.EDDiscoveryController.DoRefreshHistory(RefreshWorkerArgs args) in C:\Code\EDDiscovery\EDDiscovery\EDDiscoveryController.cs:line 811
